base/compiler/tfuncs.jl

```diff
@@ -1558,6 +1558,8 @@ function tuple_tfunc(argtypes::Vector{Any})
                 else
                     params[i] = Type
                 end
+            elseif x === DataType || x === UnionAll
```

Comment:
```suggestion
            elseif iskindtype(x)
```
or somewhat equivalently `isconcretetype(x)`?

Reply:
Thanks, I think `iskindtype` tells the purpose of this special case more explicitly.